src/main/java/org/camunda/community/benchmarks/JobWorker.java

```diff
@@ -22,32 +22,37 @@
 import io.camunda.zeebe.spring.client.jobhandling.CommandWrapper;
 
 @Component
-public class JobWorker {
+public class JobWorker implements ApplicationListener<ContextRefreshedEvent> {
     private static final Logger LOG = LogManager.getLogger(JobWorker.class);
 
-    @Autowired
-    private BenchmarkConfiguration config;
+    private final BenchmarkConfiguration config;
 
-    @Autowired
-    private BenchmarkCompleteJobExceptionHandlingStrategy exceptionHandlingStrategy;
+    private final BenchmarkCompleteJobExceptionHandlingStrategy exceptionHandlingStrategy;
 
     // TODO: Check if we can/need to check if the scheduler can catch up with all its work (or if it is overwhelmed)
-    @Autowired
-    private TaskScheduler scheduler;
+    private final TaskScheduler scheduler;
+
+    private final ZeebeClient client;
 
-    @Autowired
-    private ZeebeClient client;
+    private final StatisticsCollector stats;
 
-    @Autowired
-    private StatisticsCollector stats;
+    public JobWorker(BenchmarkConfiguration config,
+        BenchmarkCompleteJobExceptionHandlingStrategy exceptionHandlingStrategy,
+        @Qualifier("taskScheduler") TaskScheduler scheduler, ZeebeClient client, StatisticsCollector stats) {
+        this.config = config;
+        this.exceptionHandlingStrategy = exceptionHandlingStrategy;
+        this.scheduler = scheduler;
+        this.client = client;
+        this.stats = stats;
+    }
 
     private void registerWorker(String jobType) {
 
         long fixedBackOffDelay = config.getFixedBackOffDelay();
 
         JobWorkerBuilderStep1.JobWorkerBuilderStep3 step3 = client.newWorker()
                 .jobType(jobType)
-                .handler(new SimpleDelayCompletionHandler(false))
+                .handler(new SimpleDelayCompletionHandler(true))
```

Comment:
what does this flag do?

Reply:
it's to markProcessInstanceCompleted, so the app will create metric over all completed instance